crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -650,58 +646,42 @@ async fn run_validation_callback(
         ValidateResult::Valid => Ok(Outcome::Accepted),
         ValidateResult::Invalid(reason) => Ok(Outcome::Rejected(reason)),
         ValidateResult::UnresolvedDependencies(UnresolvedDependencies::Hashes(hashes)) => {
+            tracing::debug!(
+                ?hashes,
+                "Op validation returned unresolved dependencies -  AgentActivity"
+            );
             // fetch all missing hashes in the background without awaiting them
             let cascade_workspace = workspace.clone();
             let cascade =
                 CascadeImpl::from_workspace_and_network(&cascade_workspace, network.clone());
 
-            // add hashes missing to validate an op to hash map
-            hashes.iter().for_each(|hash| {
-                validation_dependencies
-                    .lock()
-                    .insert_hash_missing_for_op(dht_op_hash.clone(), hash.clone());
-            });
-
-            // build a collection of futures to fetch the individual missing
-            // hashes
-            let validation_deps = validation_dependencies.clone();
-            let dht_op_hash = dht_op_hash.clone();
-            let fetches = hashes
-                .clone()
-                .into_iter()
-                .filter(move |hash| {
-                    // keep track of which dependencies are being fetched to
```

Comment:
... but may have been filtered out in case the hash was already being fetched...